internal/pkg/sourcetype/sourcetype.go

```diff
@@ -26,14 +26,16 @@ import (
 
 // IsSourceType determines whether a Type is a Source Type.
 // A Source Type is either:
-// - A Named Type that is classified as a Source
-// - A composite type that contains a Source Type
+// - A Named Struct Type that is configured as a Source
 // - A Struct Type that contains a tagged field
+// - A composite type that contains a Source Type
 func IsSourceType(c *config.Config, tf fieldtags.ResultType, t types.Type) bool {
 	deref := utils.Dereference(t)
 	switch tt := deref.(type) {
 	case *types.Named:
-		return c.IsSourceType(utils.DecomposeType(tt)) || IsSourceType(c, tf, tt.Underlying())
+		// only named structs can be *configured* sources
```

Comment:
Is this an assertion we really want to make?

Reply:
Good call, I believe the answer is no. I've reworked this snippet to deny interfaces rather than only accepting structs. Indeed, the following is a named basic type, which could also be configured as a source:

```go
type Password string
```